retriever/bitbucketretriever/retriever.go

```diff
@@ -0,0 +1,103 @@
+qpackage bitbucketretriever
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thomaspoignant/go-feature-flag/internal"
+	"github.com/thomaspoignant/go-feature-flag/retriever/shared"
+)
+
+type Retriever struct {
+	RepositorySlug     string
+	Workspace          string
+	FilePath           string
+	Commit             string
```

Comment:
**suggestion:** If I am correct `Commit` here is used to specify which branch we are using _(this is what I have tested)_. I'll suggest to rename this field `Branch` instead of `Commit` that can be misleading when configuring the retriever.

**suggestion:**  We should also try to have a default value for the `Branch` that should be `main` if not set.